connector/signaltometricsconnector/config/config.go

```diff
@@ -299,6 +318,19 @@ func validateMetricInfo[K any](mi MetricInfo, parser ottl.Parser[K]) error {
 			return fmt.Errorf("failed to parse value OTTL expression for summary: %w", err)
 		}
 	}
+	if mi.Gauge != nil {
+		metricsDefinedCount++
+		if _, err := parser.ParseValueExpression(mi.Gauge.Value); err != nil {
+			return fmt.Errorf("failed to parse value OTTL expression for gauge: %w", err)
+		}
+		// if ExtractGrokPatterns is used, validate the key selector
```

Comment:
Why only this specific ottl function is checked when any kind of OTTL function is allowed? Same check should be applied for [extractpatterns](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/ottlfuncs/README.md#extractpatterns) function, right?

I wonder if validation makes sense here since any kind of OTTL function and I'm not sure if we know before hand which of those should need validation or not.

Reply:
I am also on the fence about the need to have such validation, we had a similar discussion [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40113#discussion_r2093058092) which resulted in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/40118